core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -6004,10 +6013,13 @@ enum DefaultConstants NADefaults::holdOrRestore	(const char *attrName,
   char * value = NULL;
   if (holdOrRestoreCQD == 1) // hold cqd
     {
-      if (heldDefaults_[attrEnum])
-	{
-	  NADELETEBASIC(heldDefaults_[attrEnum], NADHEAP);
-	}
+      if (heldHeldDefaults_[attrEnum])
+        {
+          // Gasp! We've done three successive HOLDs... it's off to
+          // the bit bucket for the deepest value
+          NADELETEBASIC(heldHeldDefaults_[attrEnum], NADHEAP);
```

Comment:
I considered this when developing the fix and decided against it. A person can enter CQD HOLD commands at a sqlci prompt. With an assert, sqlci would abend when a person did three such CQDs for the same default in succession. Having user input cause an assert didn't seem good to me. Another possibility is to raise an error or a warning. I will consider tihs for a later check-in.